src/Datagrid/ProxyQuery.php

```diff
@@ -201,6 +202,22 @@ public function execute(array $params = [], $hydrationMode = null)
         }
 
         $query = $this->getDoctrineQuery();
+        $queryBuilder = $this->getQueryBuilder();
+
+        $identifierFieldNames = $this
+            ->getQueryBuilder()
+            ->getEntityManager()
+            ->getMetadataFactory()
+            ->getMetadataFor(current($this->getQueryBuilder()->getRootEntities()))
+            ->getIdentifierFieldNames();
+        $hasSingleIdentifierName = 1 === \count($identifierFieldNames);
+        $hasNoJoins = 0 === \count($queryBuilder->getDQLPart('join'));
```

Comment:
Might be better to create a variable `$hasJoin` instead.
This will be similar to `hasHavingPart`. and this avoid `!$hasNoJoin` which is a double negation.